syntax = "proto3";
package submission;

message Submission {
    string public_uuid = 1;
    string version_uuid = 2;
    string student_public_uuid = 3;
    string assignment_version_uuid = 4;
    string submission_file_group_uuid = 5;
    uint64 version_timestamp = 6;
}

message SubmissionFileGroup {
    string uuid = 1;
    string student_public_uuid = 2;
    repeated FileContent file_contents = 3;
}

message FileContent {
    string file_name = 1;
    string file_type = 2;
    bytes content = 3;
}